src/lenses/CrvRefundLens.sol

```diff
@@ -0,0 +1,153 @@
+    address private constant YBRIBE_V3_ADDR = 0x03dFdBcD4056E2F92251c7B07423E1a33a7D3F6d;
+    address private constant CURVE_MIM_GAUGE_ADDR = 0xd8b712d29381748dB89c36BCa0138d7c75866ddF;
+    address private constant CURVE_GAUGE_CONTROLLER_ADDR = 0x2F50D538606Fa9EDD2B11E2446BEb18C9D5846bB;
+    address private constant SPELL_ORACLE_ADDR = 0x75e14253dE6a5c2af12d5f1a1EA0A2E11e69EC10;
+    address private constant VE_CRV_ADDR = 0x5f3b5DfEb7B28CDbD7FAba78963EE202a494e2A2;
+    address private constant DEGENBOX = 0xd96f48665a1410C0cd669A88898ecA36B9Fc2cce;
+    address private constant FEE_WITHDRAWER = 0x9cC903e42d3B14981C2109905556207C6527D482;
+    ERC20 private constant MIM = ERC20(0x99D8a9C45b2ecA8864373A26D1459e3Dff1e17F3);
+
+    uint256 constant WEEKS_IN_YEAR = 52;
+    IMarketLens private immutable marketLens;
+    IGaugeController private immutable gaugeController;
+    IVoteEscrowedCrv private immutable voteEscrowedCrv;
+    IYBribeV3 private immutable yBribeContract;
+
+    struct AmountValue {
+        uint256 amount;
+        uint256 value;
+    }
+
+    struct RefundInfo {
+        address[] cauldrons;
+        uint256 spellPrice;
+        uint256[] userBorrowAmounts;
+        uint256 userVeCrvVoted;
+        uint256 userBribesReceived;
+    }
+    uint256 constant PRECISION = 1e18;
+    uint256 constant TENK_PRECISION = 1e5;
+    uint256 constant BPS_PRECISION = 1e4;
+
+    constructor() {
+        marketLens = IMarketLens(LENS_ADDR);
+        gaugeController = IGaugeController(CURVE_GAUGE_CONTROLLER_ADDR);
+        voteEscrowedCrv = IVoteEscrowedCrv(VE_CRV_ADDR);
+        yBribeContract = IYBribeV3(YBRIBE_V3_ADDR);
+    }
+
+    function getUserBorrowAmounts(ICauldronV4[] calldata cauldrons, address user) public view returns (uint256[] memory borrows) {
+        borrows = new uint256[](cauldrons.length);
+        for (uint256 i = 0; i < cauldrons.length; i++) {
+            borrows[i] = marketLens.getUserBorrow(address(cauldrons[i]), user);
+        }
+    }
+
+    function getVoterMimGaugeVotes(address votingAddress) public view returns (uint256) {
+        uint256 voterVeCrv = getVoterVeCrv(votingAddress);
+        uint256 mimGaugePower = getVoterMimGaugePower(votingAddress);
+        return (voterVeCrv * mimGaugePower) / 1e4;
+    }
+
+    function getVoterVeCrv(address votingAddress) public view returns (uint256) {
+        return voteEscrowedCrv.balanceOf(votingAddress);
+    }
+
+    // Get MIM gauge power of a voter. Power represents % of veCrv balance applied to gauge.
+    function getVoterMimGaugePower(address votingAddress) public view returns (uint256) {
+        (, uint256 power, ) = gaugeController.vote_user_slopes(votingAddress, CURVE_MIM_GAUGE_ADDR);
+        return power;
+    }
+
+    function getSpellPrice() public view returns (uint256) {
+        IOracle oracle = IOracle(SPELL_ORACLE_ADDR);
+        bytes memory data = abi.encodePacked(uint256(0));
+        return PRECISION ** 2 / oracle.peekSpot(data);
```

Comment:
is the idea to be 36 decimals here? It's more common to do `* 1e18`

Reply:
Are you saying to do PRECISION * PRECISION instead of PRECISION ** 2?